pkg/freeze/containerd/pause_test.go

```diff
@@ -0,0 +1,149 @@
+package containerd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/containerd/containerd"
+	"google.golang.org/grpc"
+	cri "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+	"knative.dev/container-freezer/pkg/daemon"
+)
+
+var (
+	containers []*cri.Container
+	results    []string
+	method     string
+)
+
+type FakeContainerdCri struct {
+	containers []*cri.Container
+	results    []string
+	method     string
+}
+
+// Container creates a CRI container with the given container ID and name
+func Container(id, name string) *cri.Container {
+	return &cri.Container{
+		Id: id,
+		Metadata: &cri.ContainerMetadata{
+			Name: name,
+		},
+	}
+}
+
+func (c *FakeContainerdCri) List(ctx context.Context, conn *grpc.ClientConn, podUID string) (*cri.ListContainersResponse, error) {
+	listContainers := &cri.ListContainersResponse{
+		Containers: c.containers,
+	}
+	containerIDs, err := lookupContainerIDs(listContainers)
+	if err != nil {
+		return nil, fmt.Errorf("expected list of containers ids but got %w", err)
+	}
+	if !reflect.DeepEqual(containerIDs, c.results) {
+		return nil, fmt.Errorf("list of containers did not match expected results: got %q, want %q", c.results, containerIDs)
+	}
+	return listContainers, nil
+}
+
+func (c *FakeContainerdCri) Pause(ctx context.Context, ctrd *containerd.Client, container string) error {
+	if c.method != "freeze" {
```

Comment:
Instead of this I think it'd be clearer to have Pause store which containers it has been called with (`c.paused = append(c.paused, container)`), then the test can assert which containers it expects Thaw to have tried to Pause (and can also assert that no containers were unexpectedly resumed)